app/charts/map/map.tsx

```diff
@@ -208,152 +208,150 @@ export const MapComponent = () => {
-                      const { observation } = d.properties;
-
-                      if (observation) {
-                        const value = areaLayer.getValue(observation);
-
-                        if (value !== null) {
-                          return areaLayer.getColor(value);
-                        }
-                      }
-
-                      return [222, 222, 222, 255];
-                    }}
-                  />
-                  <GeoJsonLayer
-                    id="shapes-mesh"
-                    data={features.areaLayer?.mesh}
-                    pickable={false}
-                    stroked={true}
-                    filled={false}
-                    extruded={false}
-                    lineWidthMinPixels={1}
-                    lineWidthMaxPixels={2}
-                    getLineWidth={100}
-                    lineMiterLimit={1}
-                    getLineColor={[255, 255, 255]}
-                  />
-                </>
-              )}
-
-              {symbolLayer.show && (
-                <ScatterplotLayer
-                  id="symbols"
-                  data={features.symbolLayer?.points}
-                  pickable={
-                    identicalLayerComponentIris ? !areaLayer.show : true
-                  }
-                  autoHighlight={true}
-                  opacity={0.7}
-                  stroked={false}
-                  filled={true}
-                  radiusUnits={"pixels"}
-                  radiusMinPixels={symbolLayer.radiusScale.range()[0]}
-                  radiusMaxPixels={symbolLayer.radiusScale.range()[1]}
-                  lineWidthMinPixels={1}
-                  getPosition={({ coordinates }: GeoPoint) => coordinates}
-                  getRadius={({ properties: { observation } }: GeoPoint) =>
-                    observation
-                      ? symbolLayer.radiusScale(
-                          symbolLayer.getValue(observation) as number
-                        )
-                      : 0
-                  }
-                  getFillColor={symbolColorRgbArray}
-                  getLineColor={[255, 255, 255]}
-                  onHover={({
-                    x,
-                    y,
-                    object,
-                  }: {
-                    x: number;
-                    y: number;
-                    object: GeoPoint;
-                  }) => {
-                    if (object) {
-                      setMapTooltipType("symbol");
-                      dispatchInteraction({
-                        type: "INTERACTION_UPDATE",
-                        value: {
-                          interaction: {
-                            visible: true,
-                            mouse: { x, y },
-                            d: object.properties.observation,
-                          },
+      {(areaLayer.data.length > 0
+        ? shapes.features !== undefined
+        : features.symbolLayer?.points.length !== undefined) && (
```

Comment:
Nit: This condition is not super clear, maybe there could be temp variables like `hasPoints`, `hasShapes`  ?

Reply:
Thanks @ptbrowne, you were right it wasn't very clear; I refactored some code in preceding steps, hopefully it is be better now :)